Snowflake and Core OAuth (#6435)
fishtown-analytics/docs.getdbt.com: website/docs/docs/core/connect-data-platform/snowflake-setup.md

Refer to the [Snowflake docs](https://docs.snowflake.com/en/sql-reference/parameters.html#label-allow-id-token) for info on how to enable this feature in your account.

### OAuth authorization

See the Snowflake [docs](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview) for information about configuring OAuth. Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.

> The OAUTH_REDIRECT_URI you provide in Snowflake is `http://localhost`. Once Snowflake is configured, add the following to your `profiles.yml` file:

Author: ```suggestion
The OAUTH_REDIRECT_URI you provide in Snowflake is `http://localhost:PORT_NUMBER`. For example, `http://localhost:8080`.

Once your Snowflake admin has configured OAuth, add the following to your `profiles.yml` file:
```

Refer to the [Snowflake docs](https://docs.snowflake.com/en/sql-reference/parameters.html#label-allow-id-token) for info on how to enable this feature in your account.

### OAuth authorization

> See the Snowflake [docs](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview) for configuring OAuth. Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.

Reviewer: Maybe make this active and say something like:
```suggestion
To learn how to configure OAuth in Snowflake, refer to their [documentation](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview). Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.
```

> See the Snowflake [docs](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview) for configuring OAuth. Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.

Author: ```suggestion
To learn how to configure OAuth in Snowflake, refer to their [documentation](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview). Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.
```

> See the Snowflake [docs](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview) for configuring OAuth. Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.

Author: ```suggestion
To learn how to configure OAuth in Snowflake, refer to their [documentation](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview). Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.

```

> See the Snowflake [docs](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview) for information about configuring OAuth. Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.

Author: ```suggestion
See the Snowflake [docs](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview) for configuring OAuth. Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.
```

> The OAUTH_REDIRECT_URI you provide in Snowflake is `http://localhost:PORT_NUMBER`. For example, `http://localhost:8080`.

Reviewer: ```suggestion
Provide the OAUTH_REDIRECT_URI in Snowflake:`http://localhost:PORT_NUMBER`. For example, `http://localhost:8080`.
```

Author: They don't provide it in our product; it's in the Snowflake docs.

> Once your Snowflake admin has configured OAuth, add the following to your `profiles.yml` file:

Reviewer: I wonder if this and the previous should be steps?

Author: This aligns with the examples in this portion of the docs. We rely on the adapter documentation for that. Important if it ever changes, then we don't have to do the heavy lifting.
